pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -813,13 +817,16 @@ func applyConnectionPool(env *model.Environment, cluster *apiv2.Cluster, setting
 
 		if settings.Tcp.MaxConnections > 0 {
 			threshold.MaxConnections = &wrappers.UInt32Value{Value: uint32(settings.Tcp.MaxConnections)}
+			enableCircuitBreakers = true
 		}
 
 		applyTCPKeepalive(env, cluster, settings)
 	}
 
-	cluster.CircuitBreakers = &v2Cluster.CircuitBreakers{
-		Thresholds: []*v2Cluster.CircuitBreakers_Thresholds{threshold},
+	if enableCircuitBreakers {
```

Comment:
AFAIK (but please verify this), you cannot disable circuit breaking by omitting it, you need to configure all fields to high values (max uint32).

Reply:
Yeah, You can not disable circuit breaking by omitting it. https://www.envoyproxy.io/docs/envoy/latest/faq/disable_circuit_breaking#is-there-a-way-to-disable-circuit-breaking

We have to build HighValue Circuit breaking struct as the default and mutate the values that user overrides - so we get the kind of disabling behaviour.

Reply:
I see, thank you!

Reply:
This whole non-disable thing is the reason why we default to envoy defaults.